pkg/component/worker/kubelet.go

```diff
@@ -170,15 +191,21 @@ func (k *Kubelet) Start(ctx context.Context) error {
 		args["--node-labels"] = strings.Join(k.Labels, ",")
 	}
 
+	nodename, err := node.GetNodename("")
+	if err != nil {
+		return fmt.Errorf("can't get hostname: %v", err)
+	}
+
+	ipv4, ipv6, err := getIPs(nodename)
+	if err == nil && ipv4 != nil && ipv6 != nil {
+		args["--node-ip"] = ipv4.String() + "," + ipv6.String()
```

Comment:
I reckon kube-router gets in trouble without this?

Reply:
For ipv6 kube-router needs that the ipv6 address is known from the node object in kubernetes. kubelet is stupid and will not set that unless you explicitly tell it to do so. It will only autodetect the ipv6 address if you disable ipv4, so if you want kubernetes to be aware of both you need to explicitly set them.

Longterm I think this should be fixed in kubelet.
ref: https://github.com/kubernetes/kubernetes/blob/2f2c1fb8116af491ed2daabc4d4a3350bf8f6fa4/pkg/kubelet/nodestatus/setters.go#L63